src/main/kotlin/no/fdk/dataservicecatalog/handler/RDFHandler.kt

```diff
@@ -1,21 +1,276 @@
 package no.fdk.dataservicecatalog.handler
 
+import no.fdk.dataservicecatalog.domain.DataService
+import no.fdk.dataservicecatalog.domain.Status
+import no.fdk.dataservicecatalog.exception.NotFoundException
 import no.fdk.dataservicecatalog.repository.DataServiceRepository
+import org.apache.jena.rdf.model.Model
+import org.apache.jena.rdf.model.ModelFactory
+import org.apache.jena.rdf.model.ResourceFactory
 import org.apache.jena.riot.Lang
+import org.apache.jena.sparql.vocabulary.FOAF
+import org.apache.jena.util.FileUtils
+import org.apache.jena.util.URIref
+import org.apache.jena.vocabulary.*
 import org.springframework.stereotype.Component
+import java.io.StringWriter
 
 @Component
-class RDFHandler(private val repository: DataServiceRepository) {
+class RDFHandler(private val repository: DataServiceRepository, private val properties: ApplicationProperties) {
 
     fun findAll(lang: Lang): String {
-        return ""
+        val dataServices = repository.findAllByStatus(Status.PUBLISHED).groupBy(DataService::catalogId)
+
+        if (dataServices.isEmpty()) return ""
```

Comment:
usikker på om dette blir gyldigt for en tom katalog for alle format, om den ikke finner noen datatjenester så kan du la jena serialisere den tomme modellen